### PR TITLE
skills for fishing, ceramics, and glassblowing (and more)

### PR DESCRIPTION
**Pull request by jjpark-kb**
## About The Pull Request
different weighting for fishing, ceramic, and glassblowing items

created a fishing skill
gives you 2x the rewards for fishing

Water basins if a bluespace crystal is used on it will "link it to a random ocean," allowing you to fish from it

fishing bobbers will now just make a sound, rather than move (the moving was not too good)
fishing bobbers will allow you to reel in at a later time as well, as it loops now (every four seconds)

created a ceramic skill
it takes less time to do ceramics

created a glassblowing skill
it takes less time to do glassblowing
glass stays hot for longer

glass lens and globe are fixed icon and desc

glassblowing items are sold for three times more to balance it against ceramics

## How This Contributes To The Skyrat Roleplay Experience
QoL for fishing, ceramics, and glassblowing... additionally connecting content to content

## Changelog

:cl:
add: added fishing skillchip
add: added ceramics skillchip
add: added glassblowing skillchip
add: added the ability to fish from water basins (forging) if bluespace crystal'd
balance: fishing bobbers make a sound rather than move
balance: fishing bobbers can now loop, rather than one-time only
balance: fishing, ceramics, and glassblowing items have some different weights
balance: glassblowing is sold for 3x more
fix: glass lens and globe icon and desc is fixed
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/fishing/code/fishing.dm [hunk 1]
```diff
@@ -9,6 +9,18 @@ GLOBAL_LIST_INIT(fishing_weights, list(
...
+	name = "Fishing Master skillchip"
+	desc = "A master of fishing, capable of wrangling the whole ocean if we must."
+	auto_traits = list(TRAIT_FISHING_MASTER)
+	skill_name = "Fishing Master"
+	skill_description = "Master the ability to use fish."
+	skill_icon = "certificate"
+	activate_message = "<span class='notice'>The fish and junk become far more visible beneath the surface.</span>"
+	deactivate_message = "<span class='notice'>The surface begins to cloud up, making it hard to see beneath.</span>"
+
 /turf/open
```
Comment (Iamgoofball) [inline modular_skyrat/modules/fishing/code/fishing.dm:24]: :-1: Move all of this fishing crap to be a component.

File: modular_skyrat/modules/fishing/code/fishing.dm [hunk 2]
```diff
@@ -81,18 +99,23 @@ GLOBAL_LIST_INIT(fishing_weights, list(
 	icon = 'modular_skyrat/modules/fishing/icons/fishing.dmi'
 	icon_state = "bobber"
 	var/timed_fishing
+	var/obj/item/fishing_rod/connected_rod
```
Comment (Iamgoofball) [inline modular_skyrat/modules/fishing/code/fishing.dm:102]: You need to doc this for doc comments.

File: modular_skyrat/modules/fishing/code/fishing.dm [hunk 3]
```diff
@@ -81,18 +99,23 @@ GLOBAL_LIST_INIT(fishing_weights, list(
...
 
 /obj/effect/fishing_bobber/Destroy(force)
 	STOP_PROCESSING(SSobj, src)
+	connected_rod = null
 	. = ..()
 
 /obj/effect/fishing_bobber/process(delta_time)
-	if(world.time > timed_fishing && world.time < timed_fishing + 2 SECONDS)
-		pixel_y = pick(-2,2)
+	if(world.time > timed_fishing && world.time < timed_fishing + 3 SECONDS)
```
Comment (Iamgoofball) [inline modular_skyrat/modules/fishing/code/fishing.dm:114]: Use timers. Also, component all of this.

File: modular_skyrat/modules/fishing/code/fishing.dm [hunk 4]
```diff
@@ -81,18 +99,23 @@ GLOBAL_LIST_INIT(fishing_weights, list(
...
+		playsound(src, 'sound/machines/ping.ogg', 30, TRUE)
+	if(world.time > timed_fishing + 4 SECONDS)
```
Comment (Iamgoofball) [inline modular_skyrat/modules/fishing/code/fishing.dm:116]: Use timers.

File: modular_skyrat/modules/fishing/code/fishing.dm (hunk 1 above)
Comment (Iamgoofball) [reply modular_skyrat/modules/fishing/code/fishing.dm:24]: It is not 2013. We do not define shit on base classes like this anymore.

File: modular_skyrat/modules/fishing/code/fishing.dm [hunk 5]
```diff
@@ -132,22 +155,41 @@ GLOBAL_LIST_INIT(fishing_weights, list(
...
-					new spawn_objone(get_turf(user))
-			fishing_spot = null
-			QDEL_NULL(spawned_bobber)
-			return
-		var/obj/spawn_objtwo = pickweight(GLOB.fishing_weights)
-		new spawn_objtwo(get_turf(user))
+		var/repeat_code = 1
+		if(HAS_TRAIT(user, TRAIT_FISHING_MASTER))
+			repeat_code = 2
+		for(var/multiple_times in 1 to repeat_code)
```
Comment (Iamgoofball) [inline modular_skyrat/modules/fishing/code/fishing.dm:161]: No. Clear variable names. As per the contributing.md.

File: modular_skyrat/modules/fishing/code/fishing.dm [hunk 6]
```diff
@@ -132,22 +155,41 @@ GLOBAL_LIST_INIT(fishing_weights, list(
...
+			if(fishing_spot.fishspawn_possible || water_basined)
+				var/pick_choose = rand(1, 3)
```
Comment (Iamgoofball) [inline modular_skyrat/modules/fishing/code/fishing.dm:163]: You don't need to make a variable for this.

File: modular_skyrat/modules/fishing/code/fishing.dm [hunk 7]
```diff
@@ -132,22 +155,41 @@ GLOBAL_LIST_INIT(fishing_weights, list(
...
+						var/obj/spawn_objone = pickweight(GLOB.fishing_weights)
+						new spawn_objone(get_turf(user))
+			else
+				var/obj/spawn_objtwo = pickweight(GLOB.fishing_weights)
+				new spawn_objtwo(get_turf(user))
+		water_basined = FALSE
 		fishing_spot = null
 		QDEL_NULL(spawned_bobber)
 		return
+	if(istype(target, /obj/structure/reagent_water_basin))
```
Comment (Iamgoofball) [inline modular_skyrat/modules/fishing/code/fishing.dm:177]: No. Use signals and components.

File: modular_skyrat/modules/fishing/code/fishing.dm [hunk 8]
```diff
@@ -132,22 +155,41 @@ GLOBAL_LIST_INIT(fishing_weights, list(
...
+		var/obj/structure/reagent_water_basin/target_basin = target
+		if(!target_basin.bluespaced)
+			to_chat(user, span_warning("[target_basin] is not connected to another ocean... perhaps using a bluespace crystal will do so?"))
+			return
+		var/turf/open/basin_turf = get_turf(target_basin)
+		if(get_dist(target, user) >= 4) //not using proximity because I want you to be able to cast the line
+			return
+		to_chat(user, span_notice("You throw out the line to [basin_turf]..."))
+		var/choose_timer = (rand(3, 7)) SECONDS
```
Comment (Iamgoofball) [inline modular_skyrat/modules/fishing/code/fishing.dm:186]: # no. 

use timers.

File: modular_skyrat/modules/primitive_fun/code/glassblowing.dm [hunk 9]
```diff
@@ -142,7 +161,9 @@
 
 /obj/item/glassblowing/blowing_rod/attackby(obj/item/I, mob/living/user, params)
 	var/obj/item/glassblowing/molten_glass/find_glass = locate() in contents
-
+	var/actions_time = 5 SECONDS
```
Comment (Iamgoofball) [inline modular_skyrat/modules/primitive_fun/code/glassblowing.dm:164]: Defines. Use defines.

File: modular_skyrat/modules/reagent_forging/code/modules/reagent_forging/forge.dm [hunk 10]
```diff
@@ -501,7 +501,10 @@
 			to_chat(user, span_warning("You feel that setting [ceramic_item] would not yield anything useful!"))
 			return
 		to_chat(user, span_notice("You start setting [ceramic_item]..."))
-		if(!do_after(user, 5 SECONDS, target = src))
+		var/set_time = 5 SECONDS
```
Comment (Iamgoofball) [inline modular_skyrat/modules/reagent_forging/code/modules/reagent_forging/forge.dm:504]: Use defines. And Trinary operators.

File: modular_skyrat/modules/reagent_forging/code/modules/reagent_forging/forge.dm [hunk 11]
```diff
@@ -525,13 +528,20 @@
 			to_chat(user, span_warning("[blowing_item] does not have any glass to heat up."))
 			in_use = FALSE
 			return
+		var/looping_amount = 5
+		var/looping_time = 2 SECONDS
+		var/added_molten = 25 SECONDS
+		if(HAS_TRAIT(user, TRAIT_GLASSBLOWING_MASTER))
+			looping_amount = 3
+			looping_time = 1 SECONDS
+			added_molten = 40 SECONDS
```
Comment (Iamgoofball) [inline modular_skyrat/modules/reagent_forging/code/modules/reagent_forging/forge.dm:537]: d e f i n e s

File: modular_skyrat/modules/reagent_forging/code/modules/reagent_forging/water_basin.dm [hunk 12]
```diff
@@ -6,12 +6,24 @@
...
+		if(bluespaced)
+			to_chat(user, span_warning("[src] has already been connected to another ocean, it does not require this again!"))
+			return
+		var/obj/item/stack/try_stack = I
+		if(!try_stack.use(1))
+			to_chat(user, span_warning("[try_stack] is unable to be added to [src], there is an issue!"))
+			return
+		bluespaced = TRUE
+		return
 	if(istype(I, /obj/item/forging/tongs))
```
Comment (Iamgoofball) [inline modular_skyrat/modules/reagent_forging/code/modules/reagent_forging/water_basin.dm:27]: All of this needs converted to components.

File: modular_skyrat/modules/reagent_forging/code/modules/reagent_forging/water_basin.dm [hunk 13]
```diff
@@ -6,12 +6,24 @@
...
 		icon_state = "primitive_water_basin"
 
 /obj/structure/reagent_water_basin/attackby(obj/item/I, mob/living/user, params)
+	if(istype(I, /obj/item/stack/ore/bluespace_crystal))
+		if(bluespaced)
+			to_chat(user, span_warning("[src] has already been connected to another ocean, it does not require this again!"))
+			return
+		var/obj/item/stack/try_stack = I
+		if(!try_stack.use(1))
+			to_chat(user, span_warning("[try_stack] is unable to be added to [src], there is an issue!"))
```
Comment (Iamgoofball) [inline modular_skyrat/modules/reagent_forging/code/modules/reagent_forging/water_basin.dm:23]: Terrible error message, please make this clarify why.

**Review by Iamgoofball — CHANGES_REQUESTED** [review]
# This code is not being merged in this state. Go learn how components and signals work first.

:-1:

**Comment by Iamgoofball** [comment]
Here is the official readme on how to component.
https://github.com/tgstation/tgstation/blob/master/code/datums/components/README.md

File: modular_skyrat/modules/fishing/code/fishing.dm (hunk 1 above)
Comment (Iamgoofball) [reply modular_skyrat/modules/fishing/code/fishing.dm:24]: https://github.com/tgstation/tgstation/blob/master/code/datums/components/README.md

**Comment by jjpark-kb** [comment]
you know what, I'm good.

**Pull request closed without merging**